src/yadism/esf/tmc.py

```diff
@@ -479,7 +537,67 @@ def _get_result_exact(self):
         return self._factor_shifted * F3out + self._factor_h3 * h3out
 
 
-ESFTMCmap = {"F2": ESFTMC_F2, "FL": ESFTMC_FL, "F3": ESFTMC_F3}
+class ESFTMC_g1(EvaluatedStructureFunctionTMC):
+    """
+    This function implements the actual formula for target mass corrections
+    for parity conserving polarized structure function g1, for the two kinds
+    described in the parent class :py:class:`EvaluatedStructureFunctionTMC`.
+
+    The formula in question can be found in :cite:`Accardi:2008pc`,
+    :cite:`Khanpour:2017cha`, and references therein.
+
+    Parameters
+    ----------
+    SF : StructureFunction
+        the interface object representing the structure function kind he
+        belongs to
+    kinematics : dict
+        requested kinematic point
+
+    """
+
+    def __init__(self, SF, kinematics):
+        super().__init__(SF, kinematics)
+        # Kinematic factor for ZM g1
+        self._factor_shifted = self.x**2 / (self.xi * self.rho**3)
+        # Kinematic factor for common for `k1` & `k2` integral
+        self._factor_k1_k2 = (4 * self.x**2 * self.mu) / self.rho**3
+        # Kinematic factor specific for `k1` integral
+        self._factor_k1 = (self.x + self.xi) / self.xi
+        # Kinematic factor specific for `k2` integral
+        self._factor_k2 = (self.rho**2 - 3) / (2 * self.rho)
+
+    def _get_result_approx(self):
+        # NOTE: The approximations are evaluated at the lower
+        # integral limit; ie by integrating the factors of `g1^0`
+        approx_k1 = 1 - self.xi
+        approx_k2 = self.xi - 1 - np.log(self.xi)
+        # collect g1 results
+        g1out = self.sf.get_esf(self.sf.obs_name, self._shifted_kinematics).get_result()
+        # Combine the expressions
+        return g1out * (
+            self._factor_shifted
+            + self._factor_k1_k2
+            * (self._factor_k1 * approx_k1 + self._factor_k2 * approx_k2)
+        )
+
+    def _get_result_exact(self):
+        # collect g1 results
+        g1out = self.sf.get_esf(self.sf.obs_name, self._shifted_kinematics).get_result()
+        # Call to the raw integrals
+        k1out = self._k1()
+        k2out = self._k2()
+        # Combine the expressions
+        return self._factor_shifted * g1out + self._factor_k1_k2 * (
+            self._factor_k1 * k1out + self._factor_k2 * k2out
+        )
+
+    def _get_result_APFEL(self):
+        raise NotImplementedError("APFEL does not implement polarized TMC")
```

Comment:
Fair enough, the name is very bad - but this actually has nothing to do with APFEL, we're just referring to the APFEL approximation, i.e. only including one integral (and not the nested)

Reply:
To be more specific looking at Eq. 20 of [0709.1775](https://arxiv.org/pdf/0709.1775.pdf) $g_2$ should also emerge from a double integral same as $k_2$ (although I can not find it straight), so here you should just drop $k_2$. (Physics wise they should refer to higher twist corrections ...)

Reply:
Hadn't noticed that you were already suggesting this! This is done done.